### PR TITLE
Enrich cs cr status

### PR DESCRIPTION
**Pull request by ZhuoxiLi**
display all installed bedrock operators' version & status in cs cr

### CODE REVIEW

File: bundle/manifests/ibm-common-service-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -42,7 +42,7 @@ spec:
   apiservicedefinitions: {}
   customresourcedefinitions:
     owned:
-      - description: CommonService is the Schema for the commonservices API. Documentation For additional details regarding install parameters check https://ibm.biz/icpfs39install. License By installing this product you accept the license terms https://ibm.biz/icpfs39license
```
Comment (horis233) [inline bundle/manifests/ibm-common-service-operator.clusterserviceversion.yaml:45]: Just reminder, we need to keep this description. They are required by ibm certification

File: controllers/goroutines/check_iam.go [hunk 2]
```diff
@@ -34,6 +37,75 @@ import (
 	util "github.com/IBM/ibm-common-service-operator/controllers/common"
 )
 
+// UpdateCsCrStatus will update cs cr status according to each bedrock operator
+func UpdateCsCrStatus(bs *bootstrap.Bootstrap) {
```
Comment (horis233) [inline controllers/goroutines/check_iam.go:41]: Let's create another fille for this new fucntion

File: controllers/common/util.go [hunk 3]
```diff
@@ -60,6 +60,12 @@ type nsMapping struct {
 	CsNs      string   `json:"map-to-common-service-namespace"`
 }
 
+type BedrockOperator struct {
```
Comment (horis233) [inline controllers/common/util.go:63]: I believe we don't to create an extra struct here

File: bundle/manifests/ibm-common-service-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (ZhuoxiLi) [reply bundle/manifests/ibm-common-service-operator.clusterserviceversion.yaml:45]: sure, I will bring them back

**Comment by ZhuoxiLi** [comment]
@horis233 @Daniel-Fan this pr is ready for final review, thanks!

File: api/v3/commonservice_types.go [hunk 4]
```diff
@@ -54,9 +54,17 @@ type Bedrockshim struct {
 	Enabled bool `json:"enabled,omitempty"`
 }
 
+// BedrockOperator maintains a list of bedrock operators
+type BedrockOperator struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
```
Comment (horis233) [inline api/v3/commonservice_types.go:60]: @ZhuoxiLi It's better to set Version and Status to `omitempty` for future extension.

File: controllers/goroutines/bedrock_operators.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
+//
```
Comment (horis233) [inline controllers/goroutines/bedrock_operators.go:1]: How about changing the name of this file to `operator_status`?

Comment (ZhuoxiLi) [reply controllers/goroutines/bedrock_operators.go:1]: sure, good idea

File: controllers/goroutines/bedrock_operators.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+		opt.Version = installCSV[strings.IndexByte(installCSV, '.')+1:]
+	}
+
+	// fetch csv
+	csv := &olmv1alpha1.ClusterServiceVersion{}
+	csvKey := types.NamespacedName{
+		Name:      installCSV,
+		Namespace: namespace,
+	}
+	csv.SetGroupVersionKind(olmv1alpha1.SchemeGroupVersion.WithKind("ClusterServiceVersion"))
```
Comment (horis233) [inline controllers/goroutines/bedrock_operators.go:104]: This line is duplicated with `csv := &olmv1alpha1.ClusterServiceVersion{}`

**Review by horis233 — APPROVED** [review]
/lgtm

**Pull request merged**